### ISSUE TITLE
Question about single prompt training

### ISSUE THREAD

**Issue by simonlevine** [lvwerra/trl#114]
Given a single fixed prompt, I'd like to optimize a causal language model to generate highly rewarded sequences. E.g., given a fixed query `x`, I want a vector of responses `**y**` such that members of this vector rank highly with a trained reward model. Basically this is like the IMDB task but given a single, fixed prompt. However, currently the PPO training using TRL does not converge. Any advice on settings/hyperparameters for this problem? Thank you

**Comment by lvwerra** [comment]
How many different responses do you expect? I could see that there might not be enough variety in the responses in a large batch and maybe training on smaller batches would help.

**Comment by younesbelkada** [comment]
@simonlevine what architecture are you using? Can you share with us the logs of your training?

**Comment by simonlevine** [comment]
Hi, thank you for the advice, I'll tune the batch size (and forward batch size). Without going into too much detail, architecture is akin to GPT (2), but trained to generate biological sequences. The prompt is about 100 characters and the response is a fixed length of about 30. I can share training logs when I have more consistent results. Thanks agaIn!

**Comment by lvwerra** [comment]
Ok, I am closing this for now - feel free to reopen if there are any news :)
